nilmtk/api.py

```diff
@@ -208,36 +209,93 @@ def train_jointly(self,clf,d):
             train=DataSet(d[dataset]['path'])
             for building in d[dataset]['buildings']:
                 print("Loading building ... ",building)
-                train.set_window(start=d[dataset]['buildings'][building]['start_time'],end=d[dataset]['buildings'][building]['end_time'])
-                train_df = next(train.buildings[building].elec.mains().load(physical_quantity='power', ac_type=self.power['mains'], sample_period=self.sample_period))
-                train_df = train_df[[list(train_df.columns)[0]]]
-                appliance_readings = []
-                
-                for appliance_name in self.appliances:
-                    appliance_df = next(train.buildings[building].elec[appliance_name].load(physical_quantity='power', ac_type=self.power['appliance'], sample_period=self.sample_period))
-                    appliance_df = appliance_df[[list(appliance_df.columns)[0]]]
-                    appliance_readings.append(appliance_df)
 
-                if self.DROP_ALL_NANS:
-                    train_df, appliance_readings = self.dropna(train_df, appliance_readings)
+                if self.use_activations:
+                    print("Getting appliance activations")
+                    for appliance_name in self.appliances:
+                        train.set_window(start=d[dataset]['buildings'][building]['start_time'],
+                                         end=d[dataset]['buildings'][building]['end_time'])
+
+                        activations = train.buildings[building].elec[appliance_name].activation_series(
```

Comment:
`Electric.activation_series()` is deprecated. Please replace all instances by `Electric.get_activations()`!

Reply:
Sure thing!